Dockerfile

```diff
@@ -2,7 +2,7 @@ ARG PYTHON_VERSION
 
 FROM python:${PYTHON_VERSION}
 
-RUN pip install hatch==1.9.4
```

Comment:
We want to keep this version pinned to ensure everyone testing with `docker-hatch` have the same exact environment to prevent issues like "it runs in mine but not yours". The docker image is not rebuilt automatically at every run. If nothing changes in the Dockerfile, it won't get rebuilt. This means some people could have a older version of hatch and get inconsistent behaviors.

tl;dr: If you need a newer version of `hatch`, pin it to a newer version.

Reply:
It seems that hatch >= 1.7.0 will now require python3.8+, which means if we want to install hatch under py3.7, hatch 1.9.4 won't work anymore. I wonder how you got this work for py 3.7 actually.

Reply:
Can you add the link? 

You can easily test hatch with different version of python using `docker-hatch` and see what works and what doesn't.

Reply:
Yes I think https://pypi.org/project/hatch/1.7.0/ is the latest hatch version that has python 3.7 has rest in peace (https://devguide.python.org/versions/). I wonder what Keras team thinks about this, maintaining python 3.7 seems to be a pain in long run.

Reply:
Let's ask the Keras team, it would simplify things greatly if we don't have to have this weird dual support of different hatch versions and command. 

I will create a thread with them and tag you.

Reply:
I created a thread here: https://chat.google.com/room/AAAAZAWC3mc/WY3msYqOvUA/WY3msYqOvUA?cls=10

Reply:
Looks like they only support 3.9+ after all. Let's drop support for 3.7 & 3.8.

Reply:
Cool! Thank you for taking actions! Let me work on the migration for hatch in this PR, and maybe we can create other PRs to update source code?

Reply:
Here is a PR to remove our GitHub check (GCB trigger) for Python 3.7: https://github.com/Kaggle/kaggleazure/pull/31120

Note: We didn't have a trigger for Python 3.8.

Reply:
SG, to use separate PRs to update source code. Thanks for tackling this diligently :)